### PR TITLE
Bugfix billing

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Issue fix in Auto PHP Licenser

### CODE REVIEW

File: app/Http/Controllers/License/LicenseController.php [hunk 1]
```diff
@@ -43,6 +43,31 @@ private function postCurl($post_url, $post_info)
...
+        $content = curl_exec($ch);
+        curl_close($ch);
+
+        return json_decode($content, true);
+    }
+
+    public function getLicensekey()
+    {
+        $url = $this->url;
+        $api_key_secret = $this->api_key_secret;
```
Comment (sandesh556) [inline app/Http/Controllers/License/LicenseController.php:64]: if you're not using API key remove this

File: app/Http/Controllers/License/LicenseController.php [hunk 2]
```diff
@@ -43,6 +43,31 @@ private function postCurl($post_url, $post_info)
...
+        $token = $this->token;
+        $getkey = $this->getCurl($url.'api/admin/viewApiKeys?token='.$token);
```
Comment (sandesh556) [inline app/Http/Controllers/License/LicenseController.php:66]: you can use $this->token directly, only create a variable if it's used in multiple places

**Review by sandesh556 — CHANGES_REQUESTED** [review]

**Review by sandesh556 — APPROVED** [review]

File: app/Http/Controllers/License/LicenseController.php [hunk 3]
```diff
@@ -43,6 +43,29 @@ private function postCurl($post_url, $post_info)
         return $result;
     }
 
+    private function getCurl($get_url)
```
Comment (sandesh556) [inline app/Http/Controllers/License/LicenseController.php:46]: add comments to this function

File: app/Http/Controllers/License/LicenseController.php [hunk 4]
```diff
@@ -43,6 +43,29 @@ private function postCurl($post_url, $post_info)
...
+        if (curl_exec($ch) === false) {
+            echo 'Curl error: '.curl_error($ch);
+        }
+        $content = curl_exec($ch);
+        curl_close($ch);
+
+        return json_decode($content, true);
+    }
+
+    public function getLicensekey()
```
Comment (sandesh556) [inline app/Http/Controllers/License/LicenseController.php:61]: add comments to this function

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Issue 1:- After saving also, valid alert message not vanishing.
image

Issue 2:- If we entered api secret and api url both invaild then in api url field alert message not showing.
image
Issue 3:- first time we entered invalid values and click on save getting alert message after that entering valid values still showing alert message.(api secret is valid)
image

**Review by SarfrazS — APPROVED** [review]
Working fine.

**Review by bindu95 — APPROVED** [review]
**BEFORE FIX**
Entering invalid license key and saving, showing success message
Screenshot 2022-05-16 at 4 05 20 PM


**AFTER FIX**
Showing alert when we are saving with incorrect license key
Screenshot 2022-05-16 at 4 07 38 PM

**Review by sandesh556 — APPROVED** [review]

**Review by RafficMohammed — APPROVED** [review]

**Pull request merged**